packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -22,11 +22,17 @@ typedef OverlayWidgetBuilder<T extends Game> = Widget Function(
   T game,
 );
 
+typedef GameBuilder<T extends Game> = T Function(BuildContext context);
+
 /// A [StatefulWidget] that is in charge of attaching a [Game] instance into the
 /// Flutter tree.
 class GameWidget<T extends Game> extends StatefulWidget {
   /// The game instance in which this widget will render
-  final T game;
+  final T? game;
```

Comment:
This could be a getter `T get game => _currentGame`, then there would be much less code changes downstream.